### PR TITLE
#20721 added validation for getting the user as a list

### LINKED ISSUE #20721: "Login As" Data Constraints
The Login As Data should be only available for someone with the Login As Role.

In addition we want to remove the UserResource v2 and put the logic only on the UserResource v1.

### PR DESCRIPTION
**Pull request by jdotcms**
1) adds a validation on loginAsData to avoid access to user without Login As Role
2) remove the v2 endpoint in favor of users/loginAsData on the v1 (deprecated method has been removed)

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/test/java/com/dotcms/rest/api/v1/user/UserResourceTest.java [hunk 1]
```diff
@@ -326,64 +362,86 @@ public void testLoginAsData() throws Exception {
...
+        final PaginationUtil paginationUtil = mock(PaginationUtil.class);
+        when(paginationUtil.getPage(request, user, filter, page, perPage )).thenReturn(responseExpected);
+
+        UserResource resource =
+                new UserResource(webResource, userAPI, siteAPI, userHelper, errorHelper, paginationUtil, roleAPI);
+        try {
 
-        Response response = userResource
-                .loginAsData(request, httpServletResponse, userFilter, true);
+            Response response = resource.loginAsData(request, httpServletResponse, filter, page, perPage);
```
Comment (nollymar) [inline dotCMS/src/test/java/com/dotcms/rest/api/v1/user/UserResourceTest.java:439]: This try-catch can be replaced by `@Test(expected=ForbiddenException.class)`

File: dotCMS/src/test/java/com/dotcms/rest/api/v1/user/UserResourceTest.java [hunk 2]
```diff
@@ -326,64 +362,86 @@ public void testLoginAsData() throws Exception {
...
-        when( roleAPI.doesUserHaveRoles(userId2, rolesId) ).thenReturn( false ) ;
-        when( roleAPI.findRoleByFQN(Role.SYSTEM + " --> " + Role.LOGIN_AS) ).thenReturn( loginAsRole ) ;
-        when( roleAPI.doesUserHaveRole(user3, loginAsRole) ).thenReturn( true ) ;
+        final UserResourceHelper userHelper  = new UserResourceHelper(userService, roleAPI, userAPI, layoutAPI, hostWebAPI,
+                userWebAPI, permissionAPI, userProxyAPI, loginService);
+        final ErrorResponseHelper errorHelper  = mock(ErrorResponseHelper.class);
+        final User user = new User();
+        final Role loginAsRole = new Role();
+        when(initDataObject.getUser()).thenReturn(user);
+        // final InitDataObject initData = webResource.init(null, httpServletRequest, httpServletResponse, true, null);
```
Comment (nollymar) [inline dotCMS/src/test/java/com/dotcms/rest/api/v1/user/UserResourceTest.java:419]: This commented line can be removed

**Review by nollymar — APPROVED** [review]

**Review by freddyucv — COMMENTED** [review]
should it have curl-test?

**Comment by jdotcms** [comment]
> should it have curl-test?

we agreed could be one or another and the case is already cover by unit and integration test

**Pull request merged**